gz-jetty: new formula

azeey opened the pull request:
See https://github.com/gazebo-tooling/release-tools/issues/1235

j-rivero commented on Formula/gz-jetty.rb:
```diff
@@ -0,0 +1,42 @@
+class GzIonic < Formula
```
```suggestion
class GzJetty < Formula
```

j-rivero commented on Formula/gz-jetty.rb:
```diff
@@ -0,0 +1,42 @@
+  include Language::Python::Virtualenv
+
+  desc "Collection of gazebo simulation software"
+  homepage "https://github.com/gazebosim/gz-ionic"
```
```suggestion
  homepage "https://github.com/gazebosim/gz-jetty"
```

j-rivero commented on Formula/gz-jetty.rb:
```diff
@@ -0,0 +1,42 @@
+  version "0.999.999-0-20250116"
+  license "Apache-2.0"
+
+  head "https://github.com/gazebosim/gz-ionic.git", branch: "main"
```
```suggestion
  head "https://github.com/gazebosim/gz-jetty.git", branch: "main"
```

j-rivero commented on Formula/gz-jetty.rb:
```diff
@@ -0,0 +1,42 @@
+  end
+
+  test do
+    assert_predicate share/"gz/gz-ionic/release_notes.md", :exist?
```
```suggestion
    assert_predicate share/"gz/gz-jetty/release_notes.md", :exist?
```

j-rivero reviewed (commented):
Some Ionic references that we probably need to update

azeey commented:
Fixed in https://github.com/osrf/homebrew-simulation/pull/2942/commits/0202924bbdf89701ccd974aa5cff50a2675f1e5d

Also updated the bump_dependency.bash script
https://github.com/gazebo-tooling/release-tools/pull/1239
